### ISSUE TITLE
On Android external url in file array don't work for this url

### ISSUE THREAD

**Issue by numerized** [EddyVerbruggen/SocialSharing-PhoneGap-Plugin#690]
I've found a special case where a certain type of url doesn't work with android (all sharing platforms) but works on ios all sharing platforms Twitter, Facebook, Email, SMS, Slack it's perfect, thanks for awesomeness ! 👍 

https://firebasestorage.googleapis.com/v0/b/emoviser.appspot.com/o/images%2Fimage-1476230096116.jpg?alt=media&token=fc949e81-1969-4dfc-b905-38c64387fac4

It's a url from firebase, it works great with iOS plugin but not with Android

Do you think you can have a lot, it's blocking my google store submission for now.

Appstore is about to accept my submission in few hours.

```
    console.log(this.imageSaved)
    var options = {
      message: 'My Last Mooz !!!!', // not supported on some apps (Facebook, Instagram)
      subject: 'Here we go !', // fi. for email
      files: [this.imageSaved], // an array of filenames either locally or remotely
      chooserTitle: 'Pick an app' // Android only, you can override the default share sheet title
    }

    var onSuccess = (result) => {
            console.log("Sharing succeeded with message: " + result);
    }

    var onError = (msg) => {
      console.log("Sharing failed with message: " + msg);
    }

    SocialSharing.shareWithOptions(options)
      .then((result) => onSuccess)
      .catch(() => onError);
  }
```

**Comment by mateunicorn** [comment]
I have met this issue too. And don't know why!

**Comment by numerized** [comment]
check this @mateunicorn https://github.com/EddyVerbruggen/SocialSharing-PhoneGap-Plugin/issues/696

I'll try it on sunday and report here.

**Comment by DeeM297** [comment]
Do you think that the PR below is related to this encodedURI / decodedURI  issue in android ?

https://github.com/EddyVerbruggen/SocialSharing-PhoneGap-Plugin/pull/715

**Comment by numerized** [comment]
Hope so :)

**Comment by mirzailhami** [comment]
Hi @numerized, have you solve this issue?

**Comment by numerized** [comment]
No the trick didn't work for me, but I'll give it another go soon.

**Comment by mirzailhami** [comment]
What is another trick you will trying to?

**Comment by albertora** [comment]
Same issue here :(

**Comment by nonren** [comment]
I'm facing the same issue now. Any solutions for this?

**Comment by numerized** [comment]
I'll dig that in the next few weeks I'm sure there's a workaround. But professional project is not giving me time to investigate...

**Comment by lluisnieto** [comment]
Need to solve this asap. I'm interested.

**Comment by lluisnieto** [comment]
Ok the only way i found to send a firebase image was converting it to a base64 string. This conversion it's easy to do, there are many examples out there.

Only that way worked.

**Comment by lluisnieto** [comment]
Finally I got a way to share images from Firebase Storage to any service. Just follow these two conditions:

Convert the image from the uri to a base64 string. You can find plenti of examples of that conversion out there.
Send the image data using shareWithOptions(). Other methods seem not to work with image data encoded that way.
I'm interested in any other updates regarding this issue. Hope that helps anyone.

**Comment by nonren** [comment]
Hi @lluisnieto, the way you did it is convert the blob into base64 string before upload to firebase?

**Comment by lluisnieto** [comment]
Yes, but remember to use shareWithOptions( ) method. It adds an extra step for the user to choose wich network want to share but it's the only way it works. ShareVia( ) did Not work.

If anybody else has info about this, i'll be pleased to read.

**Comment by carryall** [comment]
Same issue here, any solution for this ?

**Comment by mohamedmanas23** [comment]
@lluisnieto you are right....finally that's the only solution..just wondering saving base64 string would occupy more storage right?

**Comment by jamesbenns** [comment]
Any solution for this yet? :)

**Comment by CliversonObrzut** [comment]
Just faced this issue trying to share an image from firebase storage sub-folder on Instagram in an Android device. It passes all methods correctly (canShareVia, shareViaInstagram) but, during the process of creating the Instagram message it returns to my app and toast a message saying "Unable to load image". I'm wondering if any solution was presented.

**Comment by jamesbenns** [comment]
I think it's to do with how third parties interpret the URL - unfortunately the firebase download URL relies on the token in the query parameter but I'm guessing a lot of apps disregard the query parameter on download URLs as it's pretty unconventional anyway! I've changed my app to use Cloudinary which provides a normal download link (no token in query param). I can't imagine a reliable work around exists and I don't believe this is a plugin isssue! :(

**Comment by pedrovitor074** [comment]
did someone find solution for this ?

**Comment by Carlosps** [comment]
Have you guys found any solution for that?

**Comment by niojihy** [comment]
Making all objects in a bucket publicly readable

https://cloud.google.com/storage/docs/access-control/making-data-public#buckets
